### PR TITLE
Add docker stack to start emulation environment

### PR DESCRIPTION
**Pull request by ecourtois**

### CODE REVIEW

**Review by gdoffe — DISMISSED** [review]
Good job ! Love it !
Minor fixes, I already approve !

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+        libxkbfile1 \
+        libxcb-xkb1 libxcb-image0 libxcb-render-util0 libxcb-render0 libxcb-util1 \
+        libxcb-icccm4 libxcb-keysyms1 libxcb-shape0 libxkbcommon-x11-0
+#bsdextrautils groff-base libice6 libpipeline1 libsm6 libuchardet0 libxaw7 libxcb-damage0 libxcursor1 libxft2 libxmu6 libxt6 man-db xbitmaps
+
+FROM ubuntu:20.04 as cogip-firmware
+
+ENV DEBIAN_FRONTEND noninteractive
+
+RUN apt-get update && \
```
Comment (gdoffe) [inline Dockerfile:55]: you forgot to do `apt-get update` here.

Should be:
```Dockerfile
RUN apt-get update && \                                                         
    apt-get install -y \
```

This leads to error like this:
```
6.399 Get:71 http://archive.ubuntu.com/ubuntu lunar/main amd64 libxcb-render-util0 amd64 0.3.9-1build3 [10.3 kB]
6.410 E: Failed to fetch http://archive.ubuntu.com/ubuntu/pool/main/a/alsa-ucm-conf/alsa-ucm-conf_1.2.6.3-1ubuntu9.1_all.deb  404  Not Found [IP: 91.189.91.82 80]
6.410 E: Failed to fetch http://security.ubuntu.com/ubuntu/pool/main/libx/libx11/libx11-xcb1_1.8.4-2ubuntu0.2_amd64.deb  404  Not Found [IP: 91.189.91.82 80]
6.410 E: Unable to fetch some archives, maybe run apt-get update or try with --fix-missing?
6.410 Fetched 44.4 MB in 5s (9657 kB/s)
------
Dockerfile:32
--------------------
  31 |     
  32 | >>> RUN apt-get install -y \
  33 | >>>         libgl1 \
  34 | >>>         libegl1 \
  35 | >>>         libglib2.0-0 \
  36 | >>>         libxkbcommon0 \
  37 | >>>         libdbus-1-3 \
  38 | >>>         libnss3 \
  39 | >>>         libxcomposite1 \
  40 | >>>         libxdamage1 \
  41 | >>>         libxrender1 \
  42 | >>>         libxrandr2 \
  43 | >>>         libxtst6 \
  44 | >>>         libxi6 \
  45 | >>>         libasound2 \
  46 | >>>         libxkbfile1 \
  47 | >>>         libxcb-xkb1 libxcb-image0 libxcb-render-util0 libxcb-render0 libxcb-util1 \
  48 | >>>         libxcb-icccm4 libxcb-keysyms1 libxcb-shape0 libxkbcommon-x11-0
  49 |     #bsdextrautils groff-base libice6 libpipeline1 libsm6 libuchardet0 libxaw7 libxcb-damage0 libxcursor1 libxft2 libxmu6 libxt6 man-db xbitmaps
--------------------
ERROR: failed to solve: process "/bin/sh -c apt-get install -y         libgl1         libegl1         libglib2.0-0         libxkbcommon0         libdbus-1-3         libnss3         libxcomposite1         libxdamage1         libxrender1         libxrandr2         libxtst6         libxi6         libasound2         libxkbfile1         libxcb-xkb1 libxcb-image0 libxcb-render-util0 libxcb-render0 libxcb-util1         libxcb-icccm4 libxcb-keysyms1 libxcb-shape0 libxkbcommon-x11-0" did not complete successfully: exit code: 100
ERROR: Service 'monitor' failed to build : Build failed
```

**Review by gdoffe — CHANGES_REQUESTED** [review]
One bug, but still good job ;)

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+
+x-build-firmware: &build-firmware
+  build:
+    context: .
+    target: cogip-firmware
+
+x-env-common: &env-common
+  COGIP_SERVER_HOST: beacon
+  COGIP_SERVER_PORT: 8080
+  COGIP_SERVER_URL: http://beacon:8080
```
Comment (gdoffe) [inline docker-compose.yml:19]: Possible ?
```
  COGIP_SERVER_URL: http://${COGIP_SERVER_HOST}:${COGIP_SERVER_PORT}
```

Comment (gdoffe) [reply docker-compose.yml:19]: Nevermind I tested it... :(

**Comment by gdoffe** [comment]
RIOT submodule seems not good, should be: https://github.com/RIOT-OS/RIOT/commit/ae87208d2a8a220d826e8827f4d5342907ec2b02

File: Dockerfile (hunk 1 above)
Comment (ecourtois) [reply Dockerfile:55]: I did not forget, it's just a typo ;-)
Good catch!
Done

File: docker-compose.yml (hunk 2 above)
Comment (ecourtois) [reply docker-compose.yml:19]: Yes, it's not possible...

File: docs/install.md [hunk 3]
```diff
@@ -148,3 +149,52 @@ This package can be copied to the Raspberry Pi and installed to deploy the Pytho
...
 pip install cogip-tools-1.0.0.tar.gz
 ```
+
+## Docker Method
+
+### Docker Installation
+
+See [Docker installation instructions](https://docs.docker.com/engine/install/).
+
+The packages to install are: `docker-ce docker-ce-cli containerd.io docker-buildx-plugin docker-compose-plugin`
```
[the remark this answers is not in the document]
Comment (ecourtois) [reply docs/install.md:159]: I just wanted to be highlight that the compose plugin is required, not just docker.
Removed.

**Review by gdoffe — APPROVED** [review]
LGTM !

**Pull request merged**